### PR TITLE
Add deletion_protection field in cloudvmcluster resource

### PR DESCRIPTION
**Pull request by tulika-aakriti**
part of : https://github.com/hashicorp/terraform-provider-google/issues/20284
**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
oracledatabase: added `deletion_protection` field to `google_oracle_database_cloud_vm_cluster `
```

### CODE REVIEW

File: mmv1/products/oracledatabase/CloudVmCluster.yaml [hunk 1]
```diff
@@ -66,10 +72,21 @@ examples:
...
+      deletion_protection: 'true'
+    ignore_read_extra:
+      - 'deletion_protection'
     test_vars_overrides:
+      'deletion_protection': 'false'
       'project': '"oci-terraform-testing"'
       'cloud_vm_cluster_id': '"ofake-vmcluster-full"'
       'cloud_exadata_infrastructure_id': '"ofake-exadata-for-vm-full"'
+virtual_fields:
+  - name: 'deletion_protection'
```
Comment (hao-nan-li) [inline mmv1/products/oracledatabase/CloudVmCluster.yaml:84]: What about using camel case here?

**Comment by tulika-aakriti** [comment]
> 🔴 Tests failed during RECORDING mode: `TestAccOracleDatabaseCloudVmCluster_oracledatabaseCloudVmclusterBasicExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12297/artifacts/ca45b2ba-d293-4c51-9b3a-f2c84f8fc59a/build-log/recording_build/TestAccOracleDatabaseCloudVmCluster_oracledatabaseCloudVmclusterBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12297/artifacts/ca45b2ba-d293-4c51-9b3a-f2c84f8fc59a/recording/TestAccOracleDatabaseCloudVmCluster_oracledatabaseCloudVmclusterBasicExample.log)] `TestAccOracleDatabaseCloudVmCluster_oracledatabaseCloudVmclusterFullExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12297/artifacts/ca45b2ba-d293-4c51-9b3a-f2c84f8fc59a/build-log/recording_build/TestAccOracleDatabaseCloudVmCluster_oracledatabaseCloudVmclusterFullExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12297/artifacts/ca45b2ba-d293-4c51-9b3a-f2c84f8fc59a/recording/TestAccOracleDatabaseCloudVmCluster_oracledatabaseCloudVmclusterFullExample.log)]
> 
> 🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.
> 
> View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12297/artifacts/ca45b2ba-d293-4c51-9b3a-f2c84f8fc59a/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12297/artifacts/ca45b2ba-d293-4c51-9b3a-f2c84f8fc59a/recording) for each test

The error should be fixed once https://github.com/GoogleCloudPlatform/magic-modules/pull/12296  is merged.

File: mmv1/products/oracledatabase/CloudVmCluster.yaml (hunk 1 above)
Comment (tulika-aakriti) [reply mmv1/products/oracledatabase/CloudVmCluster.yaml:84]: I think this is recommend format https://googlecloudplatform.github.io/magic-modules/develop/client-side-fields/

Comment (hao-nan-li) [reply mmv1/products/oracledatabase/CloudVmCluster.yaml:84]: Ah I see, thanks for checking out.

**Review by hao-nan-li — APPROVED** [review]

**Pull request merged**